docs/analytics/apis/taxonomy-import-api.md

```diff
@@ -0,0 +1,297 @@
+| EU Residency Server | [https://data-api.eu.amplitude.com/import/csv/](https://data-api.eu.amplitude.com/graphql) |
+
+## Taxonomy import REST API
+
+To upload a new tracking plan into a separate branch called "import" you need to provide workspace ID and `main` branch ID retrieved from the previous step. Upload file using POST request encoded as `multipart/form-data`, see curl example below:
+
+### Example request
+
+```bash
+curl -v -H "Authorization: Bearer {personal-access-token}" \
+-F csv=@/Users/amplitude-dev/downloads/my-tracking-plan.csv \
+https://data-api.amplitude.com/import/csv/{workspaceId}/{mainBranchId}
+```
+
+### Response
+
+A successful request returns a `200 OK` status with a JSON body:
+
+```json
+{
+  "importType": "stargate",
+  "eventCount": 14,
+  "templateCount": 0,
+  "branchId": "5a8c4928-f090-4244-a5fc-ee856b33fc0d",
+  "branchName": "import",
+  "stagingVersionId": "6c006657-0ee7-43e6-9c1e-97b3ddaada07",
+  "currentVersionId": "f3d6b525-7916-437f-bf5b-1b8e137892ae"
+}
+```
+
+## Publish staging version changes
+
+The imported tracking plan is stored in the `staging` version of the `import` branch. It must be published before it can be merged into the `main` branch. To publish the staging version, you need to provide the branch ID and staging version ID retrieved from the previous step. The following example shows how to do this.
+
+### Example mutation
+
+```bash
+POST /graphql HTTP/1.1
+Host: data-api.amplitude.com
+Content-Type: application/json
+Authorization: Bearer {personal-access-token}
+
+{
+  "query": "mutation PublishVersion($input: PublishVersionInput!) {
+    publishVersion(input: $input) {
+      id
+      name
+      currentVersionId
+      stagingVersionId
+    }
+  }",
+  "variables": {
+    "input": {
+      "branchId": "5a8c4928-f090-4244-a5fc-ee856b33fc0d",
+      "versionId": "6c006657-0ee7-43e6-9c1e-97b3ddaada07",
+      "description": ""
+    }
+  }
+}
+```
+
+### Response
+
+A successful request returns a `200 OK` status and a JSON body. The response contains new staging version id available for the next set of changes.
+
+```json
+{
+  "data": {
+    "publishVersion": {
+      "id": "574c4710-d85d-4a6f-bc41-8d55ca9445ec",
+      "name": "main",
+      "currentVersionId": "f030d1a5-ca67-4bac-8322-334c0231422a",
+      "stagingVersionId": "f90c0ee9-80a4-443c-b1d5-25f2af9b7a2e"
+    }
+  }
+}
+```
+
+## Pull request merge GraphQL API
+
+Changes made in `import` (or any other) branch do not affect Analytics until the branch is merged back into the `main` branch. To programmatically merge branches use Branch Merge API.
+
+First, create a new pull request to merge the `import` branch into the `main` branch. Use branch IDs retrieved from the previous steps instead of branch names, `originBranchId` should contain `import` branch ID and `targetBranchId` should contain `main` branch ID.
```

Comment:
this section describes a 2 step process:

1. `CreatePullRequest`
2. `MergePullRequest`

IMO, this could be formatted in a way that makes that more clear.  Maybe before describing each step in detail, you can say something like:

```
To merge a branch's changes (from the previous step) into the main branch, 2 steps are needed:

1. Create a pull request
2. Merge the pull request

Here are example requests for accomplishing these steps.
```